base/rtlib/RTLIB.jl

```diff
@@ -0,0 +1,241 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+"""
+    RTLIB
+
+Implements the runtime library for Julia. The implementations are based on
+llvm's compiler-rt. This implementations follows the compiler-rt naming convention
+and registers the pure Julia implementation as `extern_c` so that LLVM can find them.
+
+As a secondary interface `RTLIB.convert(::Type{T}, x)` is provided.
+"""
+module RTLIB
+
+register(f::Function, rtype::ANY, argt::ANY, name::String) =
+    ccall(:jl_extern_c, Void, (Any, Any, Any, Cstring),
+          f, rtype, argt, name)
+
+# Check if relative include is available
+if isdefined(Base, :INCLUDE_STATE) && Base.INCLUDE_STATE == 1
```

Comment:
@tkelman Is that an appropriate solution? We can only `include` files relative to ourselves once `Base.INCLUDE_STATE == 2` (and I think this is the only way of testing if we are currently building the sysimg).